### ISSUE TITLE
[Feature Request] Signing new devices

### ISSUE THREAD

**Issue by gerout7** [wireapp/wire#349]
When a new device is added to a user account any previously verified conversation will become unverified.

If a user has already verified the fingerprint of another users device, it should be possible to use the device identitiy key of the verified device to sign the new device key.

This could allow a users who adds a new device, to confirm the validity of the new device key by signing it, therefore avoiding the additional out of band verification.

Example:

Alice and Bob both use Wire on their phones. During a physical meeting, the verified their fingerprints and set their respective devices to "verified".

Now Alice goes home and adds her PC as a new device to her Wire account.

The conversation between Alice and Bob now becomes unverified again, requiring both to meet again for verification.

With the described signing support, Alice could e.g. get a message on her phone that a new device has been added to her account, asking her to confirm the device fingerprint.
By doing so, the new device fingerprint will be signed by the device key already verified by Bob. When Bob opens the conversation with Alice, he will get a message about a new device from Alice being signed by the already verified device key. The conversation will stay verified.

This would add significant value to the verification process while also improving the usablility

**Comment by gerout7** [comment]
After some research, I found that this approach has already been implemented by Riot:
https://blog.riot.im/e2e-encryption-by-default-cross-signing-is-here/

I believe adapting this approach would significantly increase the useability of wire.

**Comment by ell1e** [comment]
I really hope this can be added. Otherwise the problem is that if conversation partners have too many devices and their fingerprints occasionally change (which can happen even with just new web sessions) I just end up not verifying anybody anymore and the security significantly degrades. So this has not only a usability, but a real security impact.

**Comment by raphaelrobert** [comment]
Agreed, there is room for improvement. We will tackle MLS first and then try to improve that part.
